### PR TITLE
DOC: Structured array doc update to note `dtype[name]`

### PR DESCRIPTION
**Pull request by chethanreddy123**
I have added some content in the doc section of the structured array with some code snippets.

### CODE REVIEW

**Review by rossbar — COMMENTED** [review]
I'm -1 on this as I think it's redundant given the example in lines 166-168.

**Comment by seberg** [comment]
If Ross is not in favor, I can't disagree ;).  Although, it does seem like the `dtype[name]` access is missing from the docs?  Maybe it would make sense to insert that with a single line example after the line 166-168 introduction to `.names`?

File: doc/source/user/basics.rec.rst [hunk 1]
```diff
@@ -185,6 +186,15 @@ The string representation of a structured datatype is shown in the "list of
...
 dictionary form.
 
+The dtype object can also be accessed with both keys and values individual:
+
+>>> d = np.dtype([('x', 'i8'), ('y', 'f4')])
+>>> for key in d.names:
+...     value = d[key]
+...     print(key, value)
+x int64
+y float32
```
Comment (melissawm) [inline doc/source/user/basics.rec.rst:196]: How about a middle ground: moving this snippet to line 169?

Comment (rossbar) [reply doc/source/user/basics.rec.rst:196]: IMO the example around L166 already covers this and is more idiomatic. In principle one could add a line demonstrating `d.fields` as well, which gives this info directly.

Comment (chethanreddy123) [reply doc/source/user/basics.rec.rst:196]: @rossbar buddy, can you tell me what would be the possible way that I can include this code snippet?

**Comment by rossbar** [comment]
@seberg brings up a good point, so I went ahead and updated the example to show looking up values on the dtype by name.

**Comment by seberg** [comment]
Thanks @chethanreddy123, I hope you are happy with the adaptation, and thanks Ross for following up.

**Comment by chethanreddy123** [comment]
Yes, @seberg thank you, and @rossbar Thanks for your follow-ups.

**Pull request merged**